grails-app/controllers/org/bbop/apollo/OrganismController.groovy

```diff
@@ -123,6 +126,169 @@ class OrganismController {
+                        File newFile = new File(organism.directory + File.separator + fileName)
+
+                        System.out.println("file unzip : "+ newFile.getAbsoluteFile())
+
+                        //create all non exists folders
+                        //else you will hit FileNotFoundException for compressed folder
+                        new File(newFile.getParent()).mkdirs()
+
+                        FileOutputStream fos = new FileOutputStream(newFile)
+
+                        int len;
+                        while ((len = zis.read(buffer)) > 0) {
+                            fos.write(buffer, 0, len)
+                        }
+
+                        fos.close()
+                        ze = zis.getNextEntry()
+                    }
+                    zis.closeEntry()
+                    zis.close()
+
+                    System.out.println("Done exporting file  . . . . loading ref seqs")
+                    sequenceService.loadRefSeqs(organism)
+                    System.out.println("Done loading ref seqs")
+                    if (checkOrganism(organism)) {
+                        organism.save(failOnError: true, flush: true, insert: true)
+                    }
+                } catch (e) {
+                    log.error(e)
+                }
+
+                render findAllOrganisms()
+            } else {
+                def error = [error: 'not authorized to add organism']
+                render error as JSON
+                log.error(error.error)
+            }
+        } catch (e) {
+            def error = [error: 'problem saving organism: ' + e]
+            render error as JSON
+            e.printStackTrace()
+            log.error(error.error)
+        }
+    }
+
+    @RestApiMethod(description = "Adds a track to an existing organism returning a JSON array of all organisms", path = "/organism/addTrackToOrganism", verb = RestApiVerb.POST)
+    @RestApiParams(params = [
+            @RestApiParam(name = "username", type = "email", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "password", type = "password", paramType = RestApiParamType.QUERY)
+            , @RestApiParam(name = "commonName", type = "string", paramType = RestApiParamType.QUERY, description = "A common name (or ID) used for the organism")
+            , @RestApiParam(name = "trackData", type = "string", paramType = RestApiParamType.QUERY, description = "Track data (JBrowse JSON)")
+            , @RestApiParam(name = "type", type = "string", paramType = RestApiParamType.QUERY, description = "Track type")
+            , @RestApiParam(name = "label", type = "string", paramType = RestApiParamType.QUERY, description = "Track label")
+            , @RestApiParam(name = "key", type = "string", paramType = RestApiParamType.QUERY, description = "Track key")
+            , @RestApiParam(name = "category", type = "string", paramType = RestApiParamType.QUERY, description = "(optional) Track metadata (JSON)")
```

Comment:
Yeah, I'm not sure how that is going to work.  I'll probably finish this out after the sequence piece.    I guess I was thinking you would zip your JSON directory and then you would send up the pieces you need.  I guess the advantage of doing it the other way is that you can evaluate and customize locally and then push everything up remotely.    That certainly makes it easier to do.